src/Nethermind/Nethermind.Consensus/Processing/BlockProcessor.BlockProductionTransactionPicker.cs

```diff
@@ -55,6 +55,11 @@ public AddingTxEventArgs CanAddTransaction(Block block, Transaction currentTx, I
                 }
 
                 IReleaseSpec spec = _specProvider.GetSpec(block.Header);
+                if (currentTx.IsAboveInitCode(spec))
+                {
+                    return args.Set(TxAction.Skip, $"EIP-3860 - transaction size over max init code size");
+                }
+
```

Comment:
Question - can we do something that we won't need to duplicate this logic here?

It is partly done by TxPool, but it is not enough.

Reply:
I think this is the case for every condition. We're duplicating logic in BlockProcessor.BlockProductionTransactionPicker, TxPool, TransactionExecutor. We should refactor everything somehow. I planned to create the issue